### PR TITLE
Add test cases for AssignCommand and UnassignCommand

### LINKED ISSUE #126: Add tests for `assign` and `unassign` commands

### LINKED ISSUE #163: Enhancement: make `StudentBuilder#withModuleClasses` accept `ModuleClass`es
Currently it accepts `String`s, which is leftover from the previous `Tag` system. We can have a overload accepting `ModuleClass`, or replace the previous one if it is not needed. Same change would be needed in `SampleDataUtil#getModuleDataList`.

### PR DESCRIPTION
**Pull request by RezwanArefin01**
Resolves #126. 
Resolves #163.

### CODE REVIEW

File: src/test/java/seedu/taassist/logic/commands/UnassignCommandTest.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        // different types -> returns false
+        assertNotEquals(unassignFirstCommand, 1);
+
+        // null -> returns false
+        assertNotEquals(unassignFirstCommand, null);
+
+        // different student -> returns false
+        assertNotEquals(unassignFirstCommand, unassignSecondCommand);
+    }
+    /**
```
Comment (Bubbl3T) [inline src/test/java/seedu/taassist/logic/commands/UnassignCommandTest.java]: You might want a new linebreak here.

File: src/test/java/seedu/taassist/logic/commands/UnassignCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+     * Model stub with multiple filtered students with same assigned class.
+     * Always says it has a module.
+     */
+    private class ModelStubWithMultipleStudents extends ModelStub {
+
+        private List<Student> students = new ArrayList<>();
+
+        public ModelStubWithMultipleStudents() {
+            // ALICE and BOB with no module classes
```
Comment (Bubbl3T) [inline src/test/java/seedu/taassist/logic/commands/UnassignCommandTest.java]: ALICE and BOB should have module classes here.

File: src/test/java/seedu/taassist/logic/commands/UnassignCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+        @Override
+        public boolean hasModuleClass(ModuleClass moduleClass) {
+            return true;
+        }
+    }
+
+    @Test
+    public void equals() {
+        ModuleClass moduleClass = new ModuleClass("GEN1000");
```
Comment (Bubbl3T) [inline src/test/java/seedu/taassist/logic/commands/UnassignCommandTest.java]: I notice that the entire test case is constructed with the module class being forced to be "GEN1000". You might want to abstract it out as a constant instead, to retain better abstraction of the test cases and ease future modification.

**Review by Bubbl3T — CHANGES_REQUESTED** [review]
Only minor nits around that can improve abstraction of the program. Noticed slight inconsistency in comments that should be fixed. Good variety of test cases, well done! :)

File: src/test/java/seedu/taassist/logic/commands/UnassignCommandTest.java (hunk 3 above)
Comment (RezwanArefin01) [reply src/test/java/seedu/taassist/logic/commands/UnassignCommandTest.java]: PR #129 by @njxue adds typical module classes and module class builders. I will update this PR after that gets merged.

**Review by Berted — APPROVED** [review]
LGTM! Tests look okay and sufficient to cover most scenarios.

File: src/test/java/seedu/taassist/logic/commands/AssignCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+
+    /**
+     * Model stub with multiple filtered students. Always says it has a module.
+     */
+    private class ModelStubWithMultipleStudents extends ModelStub {
+
+        private List<Student> students = new ArrayList<>();
+
+        public ModelStubWithMultipleStudents() {
+            // ALICE and BOB with no module classes
```
Comment (Bubbl3T) [inline src/test/java/seedu/taassist/logic/commands/AssignCommandTest.java:156]:
```suggestion
            // ALICE and BOB with module classes assigned
```

**Review by Bubbl3T — CHANGES_REQUESTED** [review]
LGTM except minor nits on the test case comments.

File: src/test/java/seedu/taassist/logic/commands/AssignCommandTest.java (hunk 4 above)
Comment (RezwanArefin01) [reply src/test/java/seedu/taassist/logic/commands/AssignCommandTest.java:156]: Here ALICE and BOB has no module classes. So "no module classes" is correct. 🤔

**Review by Bubbl3T — APPROVED** [review]

**Pull request merged**